zap_dev/lib/src/generator/generator.dart

```diff
@@ -1674,6 +1680,23 @@ class _DartSourceRewriter extends GeneralizingAstVisitor<void> {
     }
   }
 
+  @override
+  void visitInterpolationElement(InterpolationElement node) {
+    if (node is InterpolationExpression) {
```

Comment:
Can we migrate to Dart 3 so we can take advantage of pattern matching? e.g.

```dart
    if (node case InterpolationExpression(:final SimpleIdentifier expression)) { ... }
```

Reply:
Absolutely! I think we can raise the SDK constraint to 3.0.0 as a minimum. If that causes other issues that need to be fixed first I can take a look at fixing those as well.

Reply:
Awesome! I opened https://github.com/simolus3/zap/pull/24 to bump the constraints for the repo and updated this PR to use the Dart 3 syntax above 👍